numpy/ma/core.py

```diff
@@ -711,14 +716,32 @@ def getdata(a, subok=True):
     array([[1, 2],
            [3, 4]])
 
+    When `dtype` is specified, this is used to process ambiguous objects:
+
+    >>> ma.getdata((1, 2, 3))
+    array([1, 2, 3])
+    >>> ma.getdata((1, 2, 3), dtype=object)
+    (1, 2, 3)
+    >>> ma.getdata((1, 2, 3), dtype=[('a', int), ('b', int), ('c', int)])
+    array((1, 2, 3), 
+      dtype=[('a', '<i4'), ('b', '<i4'), ('c', '<i4')])
     """
+    # first deal with masked arrays
     try:
         data = a._data
     except AttributeError:
-        data = np.array(a, copy=False, subok=subok)
-    if not subok:
-        return data.view(ndarray)
-    return data
+        pass
+    else:
+        if not subok:
+            data = data.view(ndarray)
+        return data
+
+    # object dtype should not be converted to an array
+    if np.dtype(dtype) == np.object_:
```

Comment:
I definitely do not want `hasobject here` - that would catch structured dtypes, whereas I specifically want just object types